lib/elixir_analyzer/exercise_test/common_checks/variable_names.ex

```diff
@@ -0,0 +1,63 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.VariableNames do
+  @moduledoc """
+  Reports the first variable with a name that's not snake_case.
+
+  Doesn't report more if there are more.
+  A single comment should be enough for the student to know what to fix.
+
+  Common check to be run on every single solution.
+  """
+
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.Comment
+
+  # @def_ops [:def, :defp, :defmacro, :defmacrop, :defguard, :defguardp]
+
+  @spec run(Macro.t()) :: [{:pass | :fail | :skip, %Comment{}}]
+  def run(ast) do
+    {_, names} = Macro.prewalk(ast, [], &traverse/2)
+    wrong_name = List.last(names)
+
+    if wrong_name do
+      wrong_name = to_string(wrong_name)
+      correct_name = to_snake_case(wrong_name)
+
+      [
+        {:fail,
+         %Comment{
+           type: :actionable,
+           comment: Constants.solution_variable_name_snake_case(),
+           params: %{
+             expected: correct_name,
+             actual: wrong_name
+           }
+         }}
+      ]
+    else
+      []
+    end
+  end
+
+  defp traverse({name, _meta, module} = ast, names) when is_atom(module) do
```

Comment:
Yes, I've been thinking about it more, and reached the same guess. But since the check is quite cheap, I wonder if there's really that much of a benefit. Not for us I would guess :)